forge-gui/res/cardsfolder/z/zevlor_elturel_exile.txt

```diff
@@ -5,7 +5,7 @@ PT:4/2
 K:Haste
 A:AB$ Effect | Cost$ 2 T | Triggers$ SpellCast | SpellDescription$ When you next cast an instant or sorcery spell that targets only a single opponent or a single permanent an opponent controls this turn, for each other opponent, choose that player or a permanent they control, copy that spell, and the copy targets the chosen player or permanent.
 SVar:SpellCast:Mode$ SpellCast | ValidCard$ Instant,Sorcery | ValidActivatingPlayer$ You | IsSingleTarget$ True | TargetsValid$ Opponent,Permanent.OppCtrl+inZoneBattlefield | Execute$ TrigRepeat | OneOff$ True | TriggerZones$ Battlefield | TriggerDescription$ When you next cast an instant or sorcery spell that targets only a single opponent or a single permanent an opponent controls this turn, for each other opponent, choose that player or a permanent they control, copy that spell, and the copy targets the chosen player or permanent.
-SVar:TrigRepeat:DB$ RepeatEach | RepeatPlayers$ OppNonTriggeredSpellAbilityTargetOrController | RepeatSubAbility$ DBCopy
+SVar:TrigRepeat:DB$ RepeatEach | RepeatPlayers$ OppNonTriggeredSpellAbilityTargetsOrController | RepeatSubAbility$ DBCopy
```

Comment:
This got broken now
Try patching `AbilityUtils.getDefinedPlayers` for "Triggered" like this (instead of checking for CardCollection):
```
 else if (c instanceof Iterable<?>) { // For merged permanent
                    o = ((Iterable<?>) c).iterator().next();
                    if (orCont && o instanceof Card) {
                        o = ((Card) o).getController();
                    } 
                }
```

Reply:
good catch

Reply:
with this tweak, Zevlor seems to be working again

Reply:
use the addPlayer function for Help, that can already support Iterable

Reply:
Zevlor case: If I target a creature with Shock, it will make a copy for the creature's controller as well

Reply:
the problem there is the OR Controller

you probably need to split it there:
```java
addPlayer(Iterables.filter(c, Card.class), 'Controller', players);
addPlayer(Iterables.filter(c, Player.class), '', players);
```

but shouldn't players be a PlayerCollection? it shouldn't be able to add multiples

Reply:
Hmm, nothing but errors as I try that and similar: "incompatible equality constraints" and cast issues

Reply:
```
                    if (orCont) {
                        addPlayer(ImmutableList.copyOf(Iterables.filter((Iterable<Object>)c, Player.class)), "", players);
                    }
                    addPlayer(ImmutableList.copyOf(Iterables.filter((Iterable<Object>)c, Card.class)), "Controller", players);
```

The cast should be ok if that works

Reply:
That seems to work!! Thank you!